flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcDynamicTableSource.java

```diff
@@ -96,28 +98,70 @@ public JdbcDynamicTableSource(
     public LookupRuntimeProvider getLookupRuntimeProvider(LookupContext context) {
         // JDBC only support non-nested look up keys
         String[] keyNames = new String[context.getKeys().length];
-        for (int i = 0; i < keyNames.length; i++) {
+
+        for (int i = 0; i < context.getKeys().length; i++) {
```

Comment:
Why do we need this change?
Since `LookupContext` is an interface there is no guarantee that `context.getKeys()` is always O(1)

Reply:
reverted this change